src/Disks/S3/DiskS3.cpp

```diff
@@ -553,18 +547,122 @@ void DiskS3::listObjects(const String & source_bucket, const String & source_pat
     } while (outcome.GetResult().GetIsTruncated());
 }
 
-void DiskS3::copyObject(const String & src_bucket, const String & src_key, const String & dst_bucket, const String & dst_key) const
+void DiskS3::copyObject(const String & src_bucket, const String & src_key, const String & dst_bucket, const String & dst_key,
+    std::optional<Aws::S3::Model::HeadObjectResult> head) const
+{
+    if (head && (head->GetContentLength() >= 5LL * 1024 * 1024 * 1024))
```

Comment:
What do you think to make this parameter configurable at disk settings level? and test new behavior setting this option to 0 value?